block-storage/src/main/scala/coop/rchain/blockstorage/dag/BlockDagKeyValueStorage.scala

```diff
@@ -120,7 +120,92 @@ final class BlockDagKeyValueStorage[F[_]: Concurrent: Log] private (
     }
   }
 
-  private def representation: F[BlockDagRepresentation[F]] =
+  /**
+    * Calculate view on the DAG
+    * @param latestMessagesToUse if none - returns view as per the most recent state of BlockDagStorage
+    *                            if block supplied - returns view including only latestMessages
+    * @return
+    */
+  private def representation(
+      latestMessagesToUse: Option[Set[BlockHash]] = None
+  ): F[BlockDagRepresentation[F]] = {
+
+    /** trims view on the DAG to get view as per provided `latest messages` */
+    def calculateView(
+        fullDag: KeyValueDagRepresentation,
+        tips: Set[BlockHash]
+    ): F[KeyValueDagRepresentation] = {
```

Comment:
`calculateView` depends on `BlockDagRepresentation` and it makes sense to be part of the trait. With that restricted version `getRepresentation(hash)` can be added as extension on `BlockDagKeyValueStorage`.

If you think it will be easier to test if `KeyValueDagRepresentation` is public I'm ok with that. Also if you don't want to modify file based version you can create another PR to remove it.

Reply:
@tgrospic can we remove file versions support both for dagstore and blockstore?

Reply:
We still have full nodes with file-based BlockStore so for now only file-based DAG store is safe to remove.

Reply:
https://github.com/rchain/rchain/pull/3363